### PR TITLE
Compatibilty with GHC-9.4

### LINKED ISSUE #283: Allow GHC-9.4

### PR DESCRIPTION
**Pull request by ulysses4ever**
Fix #283. There are two issues.

1. A bunch of Oleg's packages need to bump their bounds on `base`.
2. I added allow-newer for those in (1) in a separate branch and it failed on Windows with a mysterious linker error: https://github.com/ulysses4ever/hackage-security/runs/7934953899?check_suite_focus=true

### CODE REVIEW

**Comment by ulysses4ever** [comment]
Here are the two PR's with CI results readily available:
- https://github.com/ulysses4ever/hackage-security/pull/2 -- allow-newer; note that haskell-CI fails expectedly because it generates it's own `cabal-project`
- https://github.com/ulysses4ever/hackage-security/pull/1 -- identical to this PR

---

the windows linker failure is probably due to a change in GHC 9.4 moving to the clang tool chain.

**Comment by ulysses4ever** [comment]
Opened [a GHC ticket to inquire about the windows linking failure](https://gitlab.haskell.org/ghc/ghc/-/issues/22085).

**Comment by Bodigrim** [comment]
@ulysses4ever On Windows GHC 9.4 requires Cabal 3.8 for successful linking, Cabal 3.6 will not work.

**Comment by ulysses4ever** [comment]
Great, thank you, Bodigrim!

I bumped the cabal version, and Oleg updated the bounds. [The CI is all green now](https://github.com/ulysses4ever/hackage-security/pull/1/checks) (no allow-newer)!

File: example-client/example-client.cabal [hunk 1]
```diff
@@ -12,7 +12,7 @@ build-type:          Simple
 cabal-version:       >=1.10
 
 tested-with:
-  GHC==9.2.1, GHC==9.0.2,
+  GHC==9.4.1, GHC==9.2.1, GHC==9.0.2,
```
Comment (andreasabel) [inline example-client/example-client.cabal:15]: Can you bump 9.2.1 to 9.2.4 as well?

File: hackage-repo-tool/hackage-repo-tool.cabal [hunk 2]
```diff
@@ -23,7 +23,7 @@ homepage:            https://github.com/haskell/hackage-security
 bug-reports:         https://github.com/haskell/hackage-security/issues
 
 tested-with:
-  GHC==9.2.1, GHC==9.0.2,
+  GHC==9.4.1, GHC==9.2.1, GHC==9.0.2,
```
Comment (andreasabel) [inline hackage-repo-tool/hackage-repo-tool.cabal:26]: Ditto.

File: hackage-root-tool/hackage-root-tool.cabal [hunk 3]
```diff
@@ -16,7 +16,7 @@ build-type:          Simple
 cabal-version:       >=1.10
 
 tested-with:
-  GHC==9.2.1, GHC==9.0.2,
+  GHC==9.4.1, GHC==9.2.1, GHC==9.0.2,
```
Comment (andreasabel) [inline hackage-root-tool/hackage-root-tool.cabal:19]: Ditto.

File: hackage-security-HTTP/hackage-security-HTTP.cabal [hunk 4]
```diff
@@ -18,7 +18,7 @@ bug-reports:         https://github.com/haskell/hackage-security/issues
 build-type:          Simple
 
 tested-with:
-  GHC==9.2.1, GHC==9.0.2,
+  GHC==9.4.1, GHC==9.2.1, GHC==9.0.2,
```
Comment (andreasabel) [inline hackage-security-HTTP/hackage-security-HTTP.cabal:21]: Ditto.

File: hackage-security-curl/hackage-security-curl.cabal [hunk 5]
```diff
@@ -16,7 +16,7 @@ build-type:          Simple
 cabal-version:       >=1.10
 
 tested-with:
-  GHC==9.2.1, GHC==9.0.2,
+  GHC==9.4.1, GHC==9.2.1, GHC==9.0.2,
```
Comment (andreasabel) [inline hackage-security-curl/hackage-security-curl.cabal:19]: Etc.

**Review by andreasabel — CHANGES_REQUESTED** [review]
Thanks, good job!

Would you mind bumping CI 9.2.1 -> 9.2.4?  Then we'd have the latest minor versions of GHC consistently.

File: example-client/example-client.cabal (hunk 1 above)
Comment (ulysses4ever) [reply example-client/example-client.cabal]: With pleasure! (Just yesterday learned how to do group editing in one Emacs buffer 🤩) Done.

**Comment by andreasabel** [comment]
@ulysses4ever : Looks like you forgot `haskell-ci regenerate`.  (Happens to me all the time...)

**Comment by ulysses4ever** [comment]
@andreasabel I should confess I had no idea... But now I see that it updates 9.2.1 to 9.2.4 which indeed should have been done before. I'll try to keep in mind...

There's a problem though:

```
❯ haskell-ci regenerate
No haskell-ci.sh, skipping bash regeneration
*INFO* Generating GitHub config for testing for GHC versions: ghc-head 7.4.2 7.6.3 7.8.4 7.10.3 8.0.2 8.2.2 8.4.4 8.6.5 8.8.4 8.10.7 9.0.2 9.2.4 9.4.1
Invalid option `--osx=8.4.4'
```
What's up with `--osx`? I updated the PR with the current state after this command but should I try to fix this error?

**Comment by Bodigrim** [comment]
`haskell-ci regenerate` uses command line from `.travis.yml`. Remove it and use  `haskell-ci github cabal.project`.

**Comment by ulysses4ever** [comment]
@Bodigrim indeed, after removing `.travis.yml` the error has gone. Then, `haskell-ci github cabal.project` succeeds but doesn't change github workflow anyhow, but it already looks fine. Overall, I added one commit that just removes `.travis.yml`.

**Review by Bodigrim — APPROVED** [review]

**Review by andreasabel — APPROVED** [review]
Thanks!

**Comment by ulysses4ever** [comment]
Thanks everyone for shepherding this! Any chance for a Hackage revision?

**Comment by andreasabel** [comment]
> Any chance for a Hackage revision?

@Mikolaj is the hackage maintainer here and the first address for this.

**Comment by Mikolaj** [comment]
@ulysses4ever: you mean Hackage release? Of which packages?

BTW, great job. :)

**Comment by ulysses4ever** [comment]
@Mikolaj hackage-security. It's necessary for moving Cabal to 9.4.1. Revision with base bound bump would suffice I think.

**Comment by Mikolaj** [comment]
All right, I will revise all the changed bounds of hackage-security.

Will hackage-repo-tool in cabal CI work without revision/release for 9.4.1, though?

**Comment by ulysses4ever** [comment]
I hoped to get complete patch for cabal yesterday but something came up. I hope I can answer your question some time this week...

**Comment by Mikolaj** [comment]
I revised both hackage-security and hackage-repo-tool. I hope that should be enough.

**Comment by juhp** [comment]
I think the current hackage revision still doesn't allow Cabal-3.8.

**Comment by Mikolaj** [comment]
The version on hackage doesn't suport GHC 9.4, but it should support Cabal 3.8. The Hackage package display is fooled by conditionals, I think. It seems to only display the first branch or something like that. Please let me know if your tests show otherwise or if you require support for GHC 9.4.

**Pull request merged**